### PR TITLE
Ensure GuyBot doesn't flag words with guy terms as substrings

### LINKED ISSUE #9: Should the bot flag whole words and not substrings  of words?
Howdy! I noticed that this bot currently looks to see if an offending word is a substring of any of the words in the user's message (for instance, checking that `firstWords.some(w => w.includes(f)))` instead of something like `firstWords.includes(f)` or `firstWords.find(w => w === f)`.

This could lead to some false positives such as:

* `ambrosia` (bros)
* `salads` (lads)
* `ballads` (lads)
* `chapstick` (chaps)

Should the bot check for full words, instead of substrings? If so, I'm happy to open up a PR with some changes!

### PR DESCRIPTION
**Pull request by BenDMyers**
Resolves #9

### CODE REVIEW

**Comment by BenDMyers** [comment]
Tests are failing since I didn't account for punctuation - on it!

File: src/bot/onMessage.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default async function onMessage(message: IMessage): Promise<void> {
   const uniqueWords = message.cleanContent.toLocaleLowerCase().split(' ').length
   const lastWords = message.cleanContent.toLocaleLowerCase().split(' ').slice(uniqueWords - 5, uniqueWords )
 
-  const offendingWord = flaggedWords.find(f => firstWords.some(w => w.includes(f))) || flaggedWords.find(f => lastWords.some(w => w.includes(f)))
+  const offendingWord = flaggedWords.find(f => firstWords.includes(f) || flaggedWords.find(f => lastWords.includes(f)))
```
Comment (MichaelJolley) [inline src/bot/onMessage.ts:18]: Speaking of "offending," is there a closing parenthesis issue here? Line breaks below added for readability to find the offensive parenthesis.

```js
const offendingWord = 
  flaggedWords.find(f => firstWords.includes(f)) || 
  flaggedWords.find(f => lastWords.includes(f))
```

**Review by michaeljolley — CHANGES_REQUESTED** [review]

**Comment by MichaelJolley** [comment]
It already accounts for punctuation by lowercasing everything before the find. I think the issue is with parenthesis placement (shown in the review)

**Current:**
```js
flaggedWords.find(f => firstWords.includes(f) || flaggedWords.find(f => lastWords.includes(f)))
```

**Proposed:**
```js
flaggedWords.find(f => firstWords.includes(f)) ||  flaggedWords.find(f => lastWords.includes(f))
```

File: src/bot/onMessage.ts (hunk 1 above)
Comment (BenDMyers) [reply src/bot/onMessage.ts:18]: Ahhh, good catch.

Comment (MichaelJolley) [reply src/bot/onMessage.ts:18]: bitmoji

**Comment by BenDMyers** [comment]
I did still end up needing to filter out punctuation - it seems like previously it was being allowed by virtue of `guys` being a valid substring of `guys,` and so on, not by the lowercasing.

**Comment by BenDMyers** [comment]
The last failing test is `Should disregard old offenses`, and I'm not honestly sure how to address that one.

I did go to the assertions:

```js
createDMStub.called.should.be.true
reactStub.called.should.be.true
replyStub.called.should.be.false
saveNoticeStub.called.should.be.true
```

and I commented out all but the first one:

```js
createDMStub.called.should.be.true
// reactStub.called.should.be.true
// replyStub.called.should.be.false
// saveNoticeStub.called.should.be.true
```

And the test is failing for me locally still, in case that helps.

**Comment by MichaelJolley** [comment]
Curious if it's related to the `package-lock.json` file changing. That's a complete guess, but the only thing that makes sense based on the limited change to the actual JS.

**Comment by BenDMyers** [comment]
> Curious if it's related to the package-lock.json file changing. That's a complete guess, but the only thing that makes sense based on the limited change to the actual JS.

Ooh, good call - I'll revert those changes and see what happens 🙂

**Comment by MichaelJolley** [comment]
Heyo @BenDMyers. I fixed the issue in the unit tests. Could you merge the latest from `main` into your branch and push it up so the actions will trigger again?

**Comment by BenDMyers** [comment]
@MichaelJolley On it! I'm curious - what ended up being the issue with the tests?

**Comment by MichaelJolley** [comment]
The mock objects were using `getDate()` rather than `getTime()`.

**Review by michaeljolley — APPROVED** [review]
Thanks for finding & fixing this @BenDMyers!

**Pull request merged**